### ISSUE TITLE
How to finetune a model for sequence classification tasks using the dataset like stanfordnlp/imdb from huggingface?

### ISSUE THREAD

**Issue by JonasQN** [pytorch/torchtune#1124]
Right now, all the tutorials are focusing on finetuning the model with chat datasets; how do I prepare a sequence classification dataset for finetuning?

**Comment by ebsmothers** [comment]
Hi @JonasQN this is a good question. I think sequence classification would need some minor changes to our dataset abstractions -- I assume you'd need to feed in labels from your dataset in that case (instead of just using shifted tokens as we currently do). Also you'd need to append a head to the model that will project to some fixed number of classes. Is that the right understanding of what you're trying to do? 

If so I think something like our [text_completion_dataset](https://github.com/pytorch/torchtune/blob/88d172fd92237f69b1d18a4693695ca65ca009da/torchtune/datasets/_text_completion.py#L68) could be a good starting point, but we would need to change the labels [here](https://github.com/pytorch/torchtune/blob/88d172fd92237f69b1d18a4693695ca65ca009da/torchtune/datasets/_text_completion.py#L63) to whatever is in your dataset (depending on the format). And you would need to make sure to use a model with a classification head (we have something like this for Mistral you can use as an example [here](https://github.com/pytorch/torchtune/blob/88d172fd92237f69b1d18a4693695ca65ca009da/torchtune/models/mistral/_component_builders.py#L414)). Finally you would probably want to change [this line](https://github.com/pytorch/torchtune/blob/88d172fd92237f69b1d18a4693695ca65ca009da/recipes/full_finetune_single_device.py#L451) in the training recipe which shifts the labels (because it assumes we are doing next-token prediction). Also cc @RdoubleA here who may have more informed things to say than I do.

(Btw as you rightly point out we don't really have any tutorials or docs on how to do this. I think as we provide better support for this type of task we can also improve our documentation on how to setup your dataset for it.)

**Comment by rezadnayeri** [comment]
thank you @ebsmothers for the instruction. Could you please expand on this? I am super interested using touchtune for text classification for a 3-class problem, and having trouble doing so. Appreciate any instruction on text classification dataset builders, and also using the  reward modelling in https://github.com/pytorch/torchtune/blob/288ff4435b0cf17325b5c3b112f6859a6cdf0ea2/torchtune/models/llama2/_model_builders.py#L279,  for user selectable num_classes

also @RdoubleA as your name mentioned few times in multiple related posts, appreciate any help in this regard!

**Comment by felipemello1** [comment]
@JonasQN can we close this issue in favor of https://github.com/pytorch/torchtune/issues/1249, and have the discussion there?

**Comment by JonasQN** [comment]
> @JonasQN should we close this issue in favor of #1249, and have the discussion there?

Yeah we could, no problem for me.
